library/core/src/macros/mod.rs

```diff
@@ -105,7 +105,7 @@ macro_rules! assert_ne {
                     // The reborrows below are intentional. Without them, the stack slot for the
                     // borrow is initialized even before the values are compared, leading to a
                     // noticeable slow down.
-                    $crate::panicking::assert_failed(kind, &*left_val, &*right_val, $crate::option::Option::Some($crate::format_args!($($arg)+)));
+                    $crate::panicking::assert_failed(kind, &*left_val, &*right_val, $crate::option::Option::Some($crate::format_args!($($arg)+)), $crate::concat!("assertion failed: `", $crate::stringify!($left), " != ", $crate::stringify!($right), "`"));
```

Comment:
This means that at compile-time, assert_eq/ne will print the stringify'd form of the values, rather than Debug format them, right?

Reply:
It would always print the stringified form. no? So when it'd print `5 != 3` before, it'd now print `v != 3`, which reduces usefulness of `assert_eq`/`assert_ne` a lot.

Reply:
Yes, because we can't format the actual values in compile time unless there is a way to make `Display` const.

Reply:
But this PR requires us to always support the stringify fallback, no? That is, we won't be able to add a `~const Display` requirement to the arguments in the future, if this PR is merged.

We might still support printing via specialization though, if `~const Display` is available. Maybe, to ensure we've not painted ourselves into a corner, the stringify fallback could also be enabled via specialization on non-const contexts?

Reply:
What do you mean? This isn't and shouldn't be insta-stable, which means we can change the implementation should future developments allow const display.

Reply:
Oh right sorry I've missed the `#[rustc_const_unstable(feature = "const_assert_eq", issue = "none")]` annotation. Should also answer the question of @Mark-Simulacrum .